s3-commons/src/main/java/io/aiven/kafka/connect/s3/S3SourceBaseConfig.java

```diff
@@ -0,0 +1,449 @@
+            return properties;
+        }
+
+        final var result = new HashMap<>(properties);
+        for (final var prop : List.of(AWS_S3_PREFIX_CONFIG)) {
+            if (properties.containsKey(prop)) {
+                String template = properties.get(prop);
+                final String originalTemplate = template;
+
+                final var unitYyyyPattern = Pattern.compile("\\{\\{\\s*timestamp\\s*:\\s*unit\\s*=\\s*YYYY\\s*}}");
+                template = unitYyyyPattern.matcher(template)
+                        .replaceAll(matchResult -> matchResult.group().replace("YYYY", "yyyy"));
+
+                if (!template.equals(originalTemplate)) {
+                    LOGGER.warn("{{timestamp:unit=YYYY}} is no longer supported, "
+                            + "please use {{timestamp:unit=yyyy}} instead. " + "It was automatically replaced: {}",
+                            template);
+                }
+
+                result.put(prop, template);
+            }
+        }
+        return result;
+    }
+
+    protected static void addS3RetryPolicies(final ConfigDef configDef) {
+        var retryPolicyGroupCounter = 0;
+        configDef.define(AWS_S3_RETRY_BACKOFF_DELAY_MS_CONFIG, Type.LONG, AWS_S3_RETRY_BACKOFF_DELAY_MS_DEFAULT,
+                ConfigDef.Range.atLeast(1L), Importance.MEDIUM,
+                "S3 default base sleep time for non-throttled exceptions in milliseconds. " + "Default is "
+                        + AWS_S3_RETRY_BACKOFF_DELAY_MS_DEFAULT + ".",
+                GROUP_S3_RETRY_BACKOFF_POLICY, retryPolicyGroupCounter++, // NOPMD UnusedAssignment
+                ConfigDef.Width.NONE, AWS_S3_RETRY_BACKOFF_DELAY_MS_CONFIG);
+        configDef.define(AWS_S3_RETRY_BACKOFF_MAX_DELAY_MS_CONFIG, Type.LONG, AWS_S3_RETRY_BACKOFF_MAX_DELAY_MS_DEFAULT,
+                ConfigDef.Range.atLeast(1L), Importance.MEDIUM,
+                "S3 maximum back-off time before retrying a request in milliseconds. " + "Default is "
+                        + AWS_S3_RETRY_BACKOFF_MAX_DELAY_MS_DEFAULT + ".",
+                GROUP_S3_RETRY_BACKOFF_POLICY, retryPolicyGroupCounter++, // NOPMD UnusedAssignment
+                ConfigDef.Width.NONE, AWS_S3_RETRY_BACKOFF_MAX_DELAY_MS_CONFIG);
+        configDef.define(AWS_S3_RETRY_BACKOFF_MAX_RETRIES_CONFIG, Type.INT, S3_RETRY_BACKOFF_MAX_RETRIES_DEFAULT,
+                ConfigDef.Range.between(1L, 30), Importance.MEDIUM,
+                "Maximum retry limit " + "(if the value is greater than 30, "
+                        + "there can be integer overflow issues during delay calculation). " + "Default is "
+                        + S3_RETRY_BACKOFF_MAX_RETRIES_DEFAULT + ".",
+                GROUP_S3_RETRY_BACKOFF_POLICY, retryPolicyGroupCounter++, // NOPMD UnusedAssignment
+                ConfigDef.Width.NONE, AWS_S3_RETRY_BACKOFF_MAX_RETRIES_CONFIG);
+    }
+
+    protected static void addAwsConfigGroup(final ConfigDef configDef) {
```

Comment:
There are several configs like these which are duplicated across both source and sink configs. Can we move them to common class ?

Reply:
Unfortunately due to the validation method used in the sink common config there is a requirement to break these out which was not my initial hope. I can check to see if they can be moved into a utility class, but that was an annoying outcome of some of the validation code.

Reply:
I understand, but the whole point of this pr is to avoid duplicate code blocks in both connectors. Let's see if we can avoid them as much as possible.

Reply:
I pulled out the static methods to a common utility and they are called from there now.
It should reduce that particular issue down, a small amount of duplication is still required unfortunately but I was able to reduce it down.